core/domain/classifier_services.py

```diff
@@ -335,8 +374,86 @@ def mark_training_job_complete(job_id):
     Args:
         job_id: str. ID of the ClassifierTrainingJob.
     """
-    _update_classifier_training_job_status(job_id,
-                                           feconf.TRAINING_JOB_STATUS_COMPLETE)
+    _update_classifier_training_jobs_status(
+        [job_id], feconf.TRAINING_JOB_STATUS_COMPLETE)
+
+
+def mark_training_jobs_failed(job_ids):
+    """Updates the training job's status to failed.
+
+    Args:
+        job_ids: list(str). list of ID of the ClassifierTrainingJobs.
+    """
+    _update_classifier_training_jobs_status(
+        job_ids, feconf.TRAINING_JOB_STATUS_FAILED)
+
+
+def mark_training_job_pending(job_id):
+    """Updates the training job's status to pending.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_jobs_status(
+        [job_id], feconf.TRAINING_JOB_STATUS_PENDING)
+
+
+def _update_scheduled_check_time_for_new_training_job(job_id):
+    """Updates the next scheduled check time of job with status NEW.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(job_id))
+
+    if not classifier_training_job_model:
+        raise Exception(
+            'The ClassifierTrainingJobModel corresponding to the job_id '
+            'of the ClassifierTrainingJob does not exist.')
+
+    classifier_training_job_model.next_scheduled_check_time = (
+        datetime.datetime.utcnow() + datetime.timedelta(
+            minutes=feconf.CLASSIFIER_JOB_TTL_MINS))
+    classifier_training_job_model.put()
+
+
+def fetch_next_job():
+    """Gets next job model in the job queue.
+
+    Returns:
+        ClassifierTrainingJob. Domain object of the next training Job.
+    """
+    classifier_training_jobs = []
+    # Initially the cursor for query is set to None.
+    cursor = None
+    valid_jobs = []
+    timed_out_job_ids = []
+
+    while len(valid_jobs) == 0:
+        classifier_training_jobs, cursor, more = (
+            classifier_models.ClassifierTrainingJobModel.
+            query_new_and_pending_training_jobs(cursor))
+        for training_job in classifier_training_jobs:
+            if (training_job.status == (
+                    feconf.TRAINING_JOB_STATUS_PENDING)):
+                if (training_job.next_scheduled_check_time <= (
```

Comment:
If the VM is dead somehow and restarted after 1 hour, A request to nextJobHandler will mark all the pending jobs waiting for one hour as failed.
I think emails will be sent to you and @prasanna08 after jobs are failed and is handled in a separate issue as mention in the comments above.